packages/gatsby-theme-newrelic/src/hooks/useLoggedIn.js

```diff
@@ -0,0 +1,108 @@
+import { useContext, useEffect, useState } from 'react';
+import { LoggedInContext } from '../components/LoggedInContext';
+
+const SAVED_STATUS_KEY = 'gatsby-theme-newrelic/logged-in-status';
+const NERDGRAPH_URL = 'https://nerd-graph.service.newrelic.com/graphql';
+
+/**
+ * Asynchronously checks whether the current user is logged in.
+ * Returns an object with `loading` and `loggedIn` properties.
+ *
+ * If you have multiple components which need the user's logged in state,
+ * wrap them with `LoggedInProvider` to avoid making duplicate checks.
+ *
+ * @example
+ * ```jsx
+ * const MyComponent = () => {
+ *   const { loading, loggedIn } = useLoggedIn('io-website')
+ *
+ *   if (loading) return <div>loading...</div>
+ *   if (loggedIn) return <div>hey old friend!</div>
+ *   return <div>hey new friend!</div>
+ * }
+ * ```
+ *
+ * @param {string} serviceName
+ */
+export const useLoggedIn = (serviceName) => {
+  const savedStatus = getSavedStatus();
+  // if `context` is null, we don't have `LoggedInProvider` as
+  // an ancestor, so we'll have to make the call ourself.
+  const context = useContext(LoggedInContext);
+  const [loading, setLoading] = useState(false);
+  const [loggedIn, setLoggedIn] = useState(savedStatus);
+  const skipCall = context != null || savedStatus != null;
+
+  useEffect(() => {
+    if (skipCall) return;
```

Comment:
I think we will want chat about the `skipCall` some more.  We'll definitely want to put some type of expiration on this cookie, just to be sure this will get removed by some mechanism since logging out of the product won't clear this cookie (unless we collab with them? idk that could be something).

Reply:
the logged in state is stored in sessionStorage, so it'll get cleared whenever the user closes the current tab. i could add an expiration date to the value stored, but i think it'd be kind of overkill for this. i doubt this data will end up persisting for long enough to cause any issues

Reply:
ah, no that makes sense then